File: cpp/520.cpp
#include <string>

using namespace std;

class Solution {
public:
  bool detectCapitalUse(string word) {
    size_t cnt = 0;
    for (auto c : word)
      if (c >= 'A' && c <= 'Z')
        ++cnt;
    return cnt == word.length() || cnt == 0 ||
           (cnt == 1 && word[0] >= 'A' && word[0] <= 'Z');
  }
};

File: go/520.go
package main

func detectCapitalUse(word string) bool {
	cnt := 0
	for _, c := range word {
		if c >= 'A' && c <= 'Z' {
			cnt++
		}
	}
	return cnt == len(word) || cnt == 0 || (cnt == 1 && word[0] >= 'A' && word[0] <= 'Z')
}
